tests/integration/UploadWordLinkRecordingIntegrationTest.php

```diff
@@ -0,0 +1,239 @@
+    const RECORDING_FILE_EXTENSION = "m4a";
+    const RECORDING_FILE_NAME_1 = "WordLinks 1." . self::RECORDING_FILE_EXTENSION;
+    const RECORDING_FILE_NAME_2 = "WordLinks 2.." . self::RECORDING_FILE_EXTENSION;
+
+    /**
+     * @throws \Exception
+     * @throws GuzzleException
+     */
+    public function testCreateNewRecording(): void
+    {
+        // request payload
+        $payload = [
+            'PhoneId' => self::PHONE_ID_1,
+            'term' =>  self::TERM_1,
+            'wordLinkRecording' => [
+              'audioRecordingFilename' => self::RECORDING_FILE_NAME_1,
+              'textBackTranslation' => self::TEXT_BACK_TRANSLATION_CONTENT
+            ],
+            'Data' => base64_encode(self::AUDIO_FILE_CONTENT),
+        ];
+
+        $recordingId = $this->sendRequestAndReturnRecordingId($payload);
+
+        # verify audio recording is created correctly in the database
+        $recording = $this->getRecording($recordingId);
+
+        $this->assertEquals($payload['term'], $recording['term']);
+        $this->assertNotEmpty($recording['fileName']);
+        $this->assertEquals(self::RECORDING_FILE_EXTENSION, pathinfo($recording['fileName'], PATHINFO_EXTENSION));
+        $this->assertEquals(self::TEXT_BACK_TRANSLATION_CONTENT, $recording['textBackTranslation']);
+        $this->assertEquals($this->getProjectId($payload['PhoneId']), $recording['projectId']);
+
+        // verify auto recording is uploaded successfully
+        $uploadedRecordingFile = sprintf("%s/%s/WordLinks/%s", self::$uploadedProjectDir, $payload['PhoneId'], $recording['fileName']);
+        $this->assertFileExists($uploadedRecordingFile);
+        $this->assertEquals(self::AUDIO_FILE_CONTENT, file_get_contents($uploadedRecordingFile));
+    }
+
+    /**
+     * @throws GuzzleException
+     */
+    public function testUpdateExistingRecording()
+    {
+        $createRecordingPayload = [
+            'PhoneId' => self::PHONE_ID_2,
+            'term' => self::TERM_1,
+            'wordLinkRecording' => [
+                'audioRecordingFilename' =>self::RECORDING_FILE_NAME_1,
+                'textBackTranslation' => self::TEXT_BACK_TRANSLATION_CONTENT
+            ],
+            'Data' => base64_encode(self::AUDIO_FILE_CONTENT),
+        ];
+
+        $recordingId = $this->sendRequestAndReturnRecordingId($createRecordingPayload);
+        $createdRecording = $this->getRecording($recordingId);
+
+        // request payload
+        $updateRecordingPayload = [
+            'PhoneId' => self::PHONE_ID_2,
+            'term' => self::TERM_1,
+            'wordLinkRecording' => [
+                'audioRecordingFilename' => self::RECORDING_FILE_NAME_2,
+                'textBackTranslation' => self::TEXT_BACK_TRANSLATION_CONTENT_UPDATED
+            ],
+            'Data' => base64_encode(self::AUDIO_FILE_CONTENT_UPDATED),
+            'RecordingId' => $recordingId
```

Comment:
The update request should not require a recording id. The API should upsert based on phoneId, term.